finality-aleph/src/sync/service.rs

```diff
@@ -266,6 +282,19 @@ impl<
                     Ok(chain_event) => self.handle_chain_event(chain_event),
                     Err(e) => warn!(target: LOG_TARGET, "Error when receiving a chain event: {}.", e),
                 },
+                maybe_justification = self.justifications_from_user.next() => match maybe_justification {
+                    // If the Justification comes from user (i.e. AlephBFT), then it means we already
+                    // have received the corresponding block, and it is or soon will be imported,
+                    // independently, into the Substrate database.
+                    // Therefore, we do not create a task requesting the block from peers.
```

Comment:
Could you instead clearly state what the assumption about the justification is (i.e. "we will end up having the related block in some way and no action from sync is necessary") and add it as a usage requirement to the trait documentation? This would make it more likely that any users will be aware that they should be following this assumption when using the interface.

Reply:
sure, how about now?